include/boost/gil/extension/dynamic_image/apply_operation_base.hpp

```diff
@@ -163,6 +165,108 @@ static typename Op::result_type BOOST_FORCEINLINE apply_operation_base(const Bit
     return apply_operation_basec<Types2>(bits2,index2,detail::reduce_bind2<Types1,Bits1,Op>(bits1,index1,op));
 }
 
+#else
+
+#define GIL_APPLY_FWD_OP(z, N, text)                                                                        \
+    template <> struct apply_operation_fwd_fn<BOOST_PP_ADD(N,1)> {                                      \
+        template <typename Types, typename Bits, typename UnaryOp>                                     \
+        auto apply(Bits& bits, std::size_t index, UnaryOp op) const {        \
+            typedef typename mpl::begin<Types>::type                                             \
+            BOOST_PP_REPEAT(N, GIL_FWD_TYPEDEFS, BOOST_PP_EMPTY)                                            \
+            T##N;                                                                                       \
+            switch (index) {                                                                            \
+                BOOST_PP_REPEAT(BOOST_PP_ADD(N,1), GIL_FWD_CASE, BOOST_PP_EMPTY)                            \
+            }                                                                                           \
+            throw;                                                                                      \
+        }                                                                                               \
+        template <typename Types, typename Bits, typename UnaryOp>                                     \
+        auto applyc(const Bits& bits, std::size_t index, UnaryOp op) const { \
+            typedef typename mpl::begin<Types>::type                                             \
```

Comment:
Would it be possible to replace all the MPL stuff with MP11 (http://boost.org/libs/mp11/) ?

This is a new code and it would help if it does not introduce any more uses of MPL which I'm trying to get rid of.

Even if MP11requires a bit of temporary plumbing e.g. to map `boost::{true|false}_` to `std::{true|false}_type`, I think it is worth to try.

Reply:
I can try but this is not really new code, it's mostly copy-and-paste and getting rid of the explicit return type, letting C++14 return type deduction do the work when available.